lib/Dialect/ESI/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@ add_mlir_dialect_library(MLIRESI
   LINK_LIBS PUBLIC
   MLIREDSC
   MLIRIR
+  MLIRTransforms
```

Comment:
@teqdruid the new ESIPasses.cpp wasn't linking for me locally, so I've added this. I believe that's the right thing to do here, but I'd appreciate it if you could take a look.

Reply:
Yeah, probably. Shouldn't do any harm.